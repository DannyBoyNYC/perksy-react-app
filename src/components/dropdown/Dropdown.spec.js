const React = require('react');

const Dropdown = require('./index').default;

describe('<Dropdown.* />', () => {
  let component;
  const requiredProps = {
    id: 'foo',
    options: [
      {
        label: 'Foo',
        value: 'foo'
      },
      {
        label: 'Bar',
        value: 'bar'
      }
    ]
  };

  beforeEach(() => {
    jest.mock('react-select', () => 'Select');
  });

  describe('<Dropdown.SingleSelect />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Dropdown.SingleSelect {...requiredProps} />,
        'Dropdown'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render label text if one is passed', () => {
      component.setProps({
        label: 'Select label text'
      });
      expect(component).toIncludeText('Select label text');
    });

    it('will set the default value of the select based on option value', () => {
      component = shallowUntilTarget(
        <Dropdown.SingleSelect {...requiredProps} defaultValue="bar" />,
        'Dropdown'
      );
      const barOption = requiredProps.options[1];
      expect(component.children().prop('defaultValue')).toEqual([barOption]);
    });
  });

  describe('<Dropdown.MultiSelect />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Dropdown.MultiSelect {...requiredProps} />,
        'Dropdown'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
