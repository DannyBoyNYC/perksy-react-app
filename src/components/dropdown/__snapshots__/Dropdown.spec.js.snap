// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown.* /> <Dropdown.MultiSelect /> should render correctly with all required props 1`] = `
<div
  className="Dropdown-root-0-1-1"
>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={Array []}
    instanceId="foo"
    isMulti={true}
    isSearchable={true}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": "foo",
        },
        Object {
          "label": "Bar",
          "value": "bar",
        },
      ]
    }
    styles={
      Object {
        "control": [Function],
        "placeholder": [Function],
      }
    }
    theme={[Function]}
  />
</div>
`;

exports[`<Dropdown.* /> <Dropdown.SingleSelect /> should render correctly with all required props 1`] = `
<div
  className="Dropdown-root-0-1-1"
>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={Array []}
    instanceId="foo"
    isSearchable={true}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": "foo",
        },
        Object {
          "label": "Bar",
          "value": "bar",
        },
      ]
    }
    styles={
      Object {
        "control": [Function],
        "placeholder": [Function],
      }
    }
    theme={[Function]}
  />
</div>
`;
